Tests/PackageLoadingTests/PD_6_2_LoadingTests.swift

```diff
@@ -0,0 +1,100 @@
+                        cSettings: [
+                            .enableWarning("implicit-fallthrough"),
+                            .disableWarning("unused-parameter"),
+                            .treatAllWarnings(as: .error),
+                            .treatWarning("deprecated-declarations", as: .warning),
+                        ],
+                        cxxSettings: [
+                            .enableWarning("implicit-fallthrough"),
+                            .disableWarning("unused-parameter"),
+                            .treatAllWarnings(as: .warning),
+                            .treatWarning("deprecated-declarations", as: .error),
+                        ],
+                        swiftSettings: [
+                            .treatAllWarnings(as: .error),
+                            .treatWarning("DeprecatedDeclaration", as: .warning),
+                        ]
+                    ),
+                    .target(
+                        name: "Bar",
+                        cSettings: [
+                            .enableWarning("implicit-fallthrough"),
+                            .disableWarning("unused-parameter"),
+                            .treatAllWarnings(as: .warning),
+                            .treatWarning("deprecated-declarations", as: .error),
+                        ],
+                        cxxSettings: [
+                            .enableWarning("implicit-fallthrough"),
+                            .disableWarning("unused-parameter"),
+                            .treatAllWarnings(as: .error),
+                            .treatWarning("deprecated-declarations", as: .warning),
+                        ],
+                        swiftSettings: [
+                            .treatAllWarnings(as: .warning),
+                            .treatWarning("DeprecatedDeclaration", as: .error),
+                        ]
+                    )
+                ]
+            )
+            """
+
+        let observability = ObservabilitySystem.makeForTesting()
+        try await withKnownIssue("https://github.com/swiftlang/swift-package-manager/issues/8543: there are compilation errors on Windows") {
+            let (_, validationDiagnostics) = try await PackageDescriptionLoadingTests
+                .loadAndValidateManifest(
+                    content,
+                    toolsVersion: .v6_2,
+                    packageKind: .fileSystem(.root),
+                    manifestLoader: ManifestLoader(
+                        toolchain: try! UserToolchain.default
+                    ),
+                    observabilityScope: observability.topScope
+                )
+            try expectDiagnostics(validationDiagnostics) { results in
+                results.checkIsEmpty()
+            }
+            try expectDiagnostics(observability.diagnostics) { results in
+                results.checkIsEmpty()
+            }
+        } when: {
+            isWindows
```

Comment:
note (non-blocking): In SwiftPM, we can also use `ProcessInfo.hostOperatingSystem == .windows`